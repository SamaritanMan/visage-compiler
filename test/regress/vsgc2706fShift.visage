/**
 * VSGC-2706 : AbstractBoundComprehension: NPE during execution
 *
 * Number ranges
 *
 * @test
 * @run
 */

var x = 1.0;
var z = 10.0;
 
var range = bind for (i in [x..(x+30) step z]) i;
println(range);
// same size, un-step-synced
x = 2;
println(range);
// same size, step-synced
x = 12;
println(range);
