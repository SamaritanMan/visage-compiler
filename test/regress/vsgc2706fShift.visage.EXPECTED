[ 1.0, 11.0, 21.0, 31.0 ]
[ 2.0, 12.0, 22.0, 32.0 ]
[ 12.0, 22.0, 32.0, 42.0 ]
